### ISSUE TITLE
Ability to greylist new servers

### ISSUE THREAD

**Issue by drequivalent** [mastodon/mastodon#29266]
### Pitch

Mastodon administrators should be able to use greylist policy for the servers they see for the first time.

### Motivation

As the spam attack is getting increasingly worse, we need to learn the lessons of the past federated systems like E-mail. E-mail servers use greylists to discriminate trustable servers from junk servers like spammers.

**Comment by SergSel2006** [comment]
Also not bad idea is to implement federated blacklist to cooperate in spam filtering.

**Comment by rekcuFniarB** [comment]
Greylisting is useless nowadays and popular mail services don't use it. Greylisting was invented many years ago to discard messages sent directly from scripts without MTA but that kind of spam is rare today. Also for greylisting to work activitypub needs implementation of responses like "temporary error, please retry later" for s2s interactions, that's unlikely to happen.

**Comment by uis246** [comment]
> Also not bad idea is to implement federated blacklist to cooperate in spam filtering.

\*Happy Roscomnadzor noises\*

**Comment by edoswald** [comment]
Why is this not an option? It's kind of like we're told, oh well, deal with it.

**Comment by Shanesan** [comment]
I agree with @colinstu12, not only would this put a large burden on moderation teams but that burden slows down potential growth or exposure of small instances.

The best method to prevent this _ongoing_ spam issue is to coordinate with other mastoadmins and have the ability to:
* Systematically block incoming spam waves. 
* Perhaps a "panic button" to stop federation of new instances through mentions for 24 hours.
* Perhaps Regex.
* Perhaps on-server LLM (in before AI BAD) comparisons to other messages.
* Perhaps all of the above, enabled as needed, and automatically disabled after a period of time.

**Comment by rekcuFniarB** [comment]
> The best method to prevent this _ongoing_ spam issue is to coordinate with other mastoadmins

I think spam problem can be faciliated with UI improvement. I'd divide incoming events into categories like `events from friends`, `events from subscriptions`, `events from discussions where I participated (e.g. replies)`, `events from bots` and `events from people I never interacted with`, e.t.c.  with ability to temporarily mute some of them. In this case spam will make less harm by flooding entire inbox.
